crates/trident/src/lib.rs

```diff
@@ -563,12 +563,29 @@ impl Trident {
                         debug!("There is a runtime update staged on the host");
                         if allowed_operations.has_finalize() {
                             let mut subsystems = SUBSYSTEMS.lock().unwrap();
-                            runtime_update::finalize_update(
+                            let finalize_result = runtime_update::finalize_update(
```

Comment:
seeing this chunk of double-call-runtime_update::finalize_update (here and in update.rs) makes me think it should be refactored.

Reply:
maybe finalize_update can accept ServicingType instead of the 2 bools ...

what is currently finalize_udpate can become inner_finalize or do_finalize, and finalize_udpate can call inner_finalize passing spec/old-spec as-is ... if inner_finalize fails health checks, then if ServicingType!=ManualRollback, inner_finalize can call itself with spec/old-spec reversed (to invoke rollback).